app/routes.py

```diff
@@ -1,2 +1,269 @@
+from .models.goal import Goal
+from flask import request, Blueprint, make_response, jsonify
+from sqlalchemy import desc, asc
+import datetime
+import os
+import requests
 
+
+
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+@tasks_bp.route("", methods=["POST", "GET"])
+def handle_tasks():
+    if request.method == "GET":
+        sort_query = request.args.get("sort")
+        if sort_query == "asc":
+            tasks = Task.query.order_by(Task.title)
+        elif sort_query == "desc":
+            tasks = Task.query.order_by(Task.title.desc())
+        else:
+            tasks = Task.query.all()
+
+        tasks_response = []
+        for task in tasks:
+            tasks_response.append({
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": task.is_complete()
+            })
+        return jsonify(tasks_response)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+
+        if "title" not in request_body or "description" not in request_body\
+            or "completed_at" not in request_body:
+            return ({
+                "details": "Invalid data"
+            }, 400)
+        else:
+            new_task = Task(title=request_body["title"],
+            description=request_body["description"],
+            completed_at=request_body["completed_at"])
+
+            db.session.add(new_task)
+            db.session.commit()
+
+            return make_response({
+                "task": {
+                    "id": new_task.task_id,
+                    "title": new_task.title,
+                    "description": new_task.description,
+                    "is_complete": new_task.is_complete()
+            }}, 201)
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("", 404)
+
+    if request.method == "GET":
+        if task.goal_id is None:
+            return {
+                "task": {
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": task.is_complete()
+            }}
+        if task.task_id:
+            return {
+                "task": {
+                    "id": task.task_id,
+                    "goal_id": task.goal_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": task.is_complete()
+            }}
+        else:
+            return {
+                "task": {
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": task.is_complete()
+            }}
```

Comment:
You can refactor this section to use your code block like so 
```suggestion
            return {
                "task": task.to_json()
            }
```